db/migrate/20170529142557_unify_container_definition_and_container.rb

```diff
@@ -0,0 +1,134 @@
+class UnifyContainerDefinitionAndContainer < ActiveRecord::Migration[5.0]
+  class ContainerDefinition < ActiveRecord::Base
+    has_many :container_port_configs, :class_name => 'UnifyContainerDefinitionAndContainer::ContainerPortConfig'
+    has_many :container_env_vars, :class_name => 'UnifyContainerDefinitionAndContainer::ContainerEnvVar'
+    has_one :security_context, :as => :resource
+    has_one :container, :class_name => 'UnifyContainerDefinitionAndContainer::Container'
+  end
+
+  class Container < ActiveRecord::Base
+    belongs_to :container_definition, :class_name => 'UnifyContainerDefinitionAndContainer::ContainerDefinition'
+  end
+
+  class SecurityContext < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    belongs_to :resource, :polymorphic => true
+  end
+
+  class ContainerEnvVar < ActiveRecord::Base
+    belongs_to :container_definition, :class_name => 'UnifyContainerDefinitionAndContainer::ContainerDefinition'
+  end
+
+  class ContainerPortConfig < ActiveRecord::Base
+    belongs_to :container_definition, :class_name => 'UnifyContainerDefinitionAndContainer::ContainerDefinition'
+  end
+
+  def up
+    # attributes
+    add_column :containers, :image,              :string
+    add_column :containers, :image_pull_policy,  :string
+    add_column :containers, :memory,             :string
+    add_column :containers, :cpu_cores,          :float
+    add_column :containers, :container_group_id, :bigint
+    add_column :containers, :privileged,         :boolean
+    add_column :containers, :run_as_user,        :bigint
+    add_column :containers, :run_as_non_root,    :boolean
+    add_column :containers, :capabilities_add,   :string
+    add_column :containers, :capabilities_drop,  :string
+    add_column :containers, :command,            :text
+
+    containers = Arel::Table.new(:containers)
+    definitions = Arel::Table.new(:container_definitions)
+    say_with_time("Copying over columns from container_definition to container") do
+      %w(image image_pull_policy memory cpu_cores container_group_id privileged
+         run_as_user run_as_non_root capabilities_add capabilities_drop command).each do |column|
+        join_sql = definitions.project(definitions[column.to_sym])
+                              .where(definitions[:id].eq(containers[:container_definition_id])).to_sql
+        Container.update_all("#{column} = (#{join_sql})")
+      end
+    end
+
+    Container.includes(:container_definition => [:container_port_configs, :container_env_vars, :security_context]).find_each do |container|
```

Comment:
starting on the other side (e.g. `ContainerEnvVars`) would work better.
In general, removing `belongs_to`

our suggestion from gitter:

```sql
update security_contexts 
set resource_id = (select containers.id from containers,container_definitions 
  where containers.container_definition_id = container_definitions.id 
  and container_definition_id = security_contexts.resource_id 
  and resource_type = 'ContainerDefinition')
```